### ISSUE TITLE
AWS::Lambda::EventSourceMapping.EventSourceArn property does not support AWS::Kinesis::StreamConsumer

### ISSUE THREAD

**Issue by jmullercuber** [aws-cloudformation/cfn-python-lint#1844]
*cfn-lint version: (`0.44.1`)*

*Description of issue.*

The linter fails when creating a Lambda events source mapping from a Kinesis Stream Consumer. However, both streams and stream consumers are valid sources according to the official AWS CloudFormation docs: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-eventsourcemapping.html#cfn-lambda-eventsourcemapping-eventsourcearn
> Amazon Kinesis - The ARN of the data stream or a stream consumer.

Other relevant documentation:
- **AWS::Kinesis::Stream**: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-kinesis-stream.html
- **AWS::Kinesis::StreamConsumer**: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-kinesis-streamconsumer.html

Below is a minimal CloudFormation template, demonstrating a few different syntaxes which should all be valid, as well as errors output by `cfn-lint`.

```yaml
AWSTemplateFormatVersion: 2010-09-09

Parameters:
  LambdaFunctionArn:
    Type: String

Resources:
  KinesisStream:
    Type: AWS::Kinesis::Stream
    Properties:
      ShardCount: 1

  StreamConsumer:
    Type: AWS::Kinesis::StreamConsumer
    Properties:
      ConsumerName: MyConsumer
      StreamARN: !GetAtt KinesisStream.Arn

  # Works as expected
  01EventSourceMapping:
    Type: AWS::Lambda::EventSourceMapping
    Properties:
      BatchSize: 500
      Enabled: true
      EventSourceArn: !GetAtt KinesisStream.Arn
      FunctionName: !Ref LambdaFunctionArn
      StartingPosition: LATEST

  # Linter error but should be okay.
  # syntax: !Ref StreamConsumer
  02EventSourceMapping:
    Type: AWS::Lambda::EventSourceMapping
    Properties:
      BatchSize: 500
      Enabled: true
      EventSourceArn: !Ref StreamConsumer
      FunctionName: !Ref LambdaFunctionArn
      StartingPosition: LATEST

  # Linter error but should be okay.
  # creates same mapping to stream consumer as in 02
  # syntax: !GetAtt StreamConsumer.ConsumerARN
  03EventSourceMapping:
    Type: AWS::Lambda::EventSourceMapping
    Properties:
      BatchSize: 500
      Enabled: true
      EventSourceArn: !GetAtt StreamConsumer.ConsumerARN
      FunctionName: !Ref LambdaFunctionArn
      StartingPosition: LATEST

  # Linter error but should be okay.
  # creates same mapping to main stream as in 01
  # syntax: !GetAtt StreamConsumer.StreamARN
  04EventSourceMapping:
    Type: AWS::Lambda::EventSourceMapping
    Properties:
      BatchSize: 500
      Enabled: true
      EventSourceArn: !GetAtt StreamConsumer.StreamARN
      FunctionName: !Ref LambdaFunctionArn
      StartingPosition: LATEST

```

**Output:**
```
E3008 Property "EventSourceArn" can Ref to resources of types [AWS::MSK::Cluster] at Resources/02EventSourceMapping/Properties/EventSourceArn/Ref
cfn-linter-fail-example.yml:36:7

E3008 Property "EventSourceArn" can Fn::GetAtt to a resource of types [AWS::DynamoDB::Table, AWS::Kinesis::Stream, AWS::SQS::Queue] at Resources/03EventSourceMapping/Properties/EventSourceArn/Fn::GetAtt
cfn-linter-fail-example.yml:48:7

E3008 Property "EventSourceArn" can Fn::GetAtt to a resource of types [AWS::DynamoDB::Table, AWS::Kinesis::Stream, AWS::SQS::Queue] at Resources/04EventSourceMapping/Properties/EventSourceArn/Fn::GetAtt
cfn-linter-fail-example.yml:60:7
```

**Expected Output:**

No errors should be output for the above template.

Additionally, if `!Ref`, or `!GetAtt` are improperly used within EventSourceArn, I'd expected the error message to include AWS::Kinesis::StreamConsumer as an option:
  - `E3008 Property "EventSourceArn" can Ref to resources of types [AWS::MSK::Cluster, AWS::Kinesis::StreamConsumer] at Resources/...`
  - `E3008 Property "EventSourceArn" can Fn::GetAtt to a resource of types [AWS::DynamoDB::Table, AWS::Kinesis::Stream, AWS::Kinesis::StreamConsumer, AWS::SQS::Queue] at Resources/...`

Reviewing the release history, I think this has been an issue as of [`v0.40.0`](https://github.com/aws-cloudformation/cfn-python-lint/releases/tag/v0.40.0)

**Comment by kddejong** [comment]
Fixed and released with v0.44.2

**Comment by jmullercuber** [comment]
Hi @kddejong , thank you for the swift resolution on this issue!

However one use case is still returning an error unfortunately:
```yaml
  02EventSourceMapping:
    Type: AWS::Lambda::EventSourceMapping
    Properties:
      BatchSize: 500
      Enabled: true
      EventSourceArn: !Ref StreamConsumer  # <--- error here
      FunctionName: !Ref LambdaFunctionArn
      StartingPosition: LATEST
```
**Output:**
```
E3008 Property "EventSourceArn" can Ref to resources of types [AWS::MSK::Cluster] at Resources/02EventSourceMapping/Properties/EventSourceArn/Ref
template.yml:36:7
```

Would it be possible to re-open this issue until that is resolved?

**Comment by jmullercuber** [comment]
Thank you for getting this resolved! I just saw that this got released in v0.48.2 ^_^
